File: 04-functions/py/fizzbuzz.py
if __name__ == "__main__":
    
    def fizzbuzz(x):
        if x % 15 == 0:
            print("FizzBuzz")
        elif x % 3 == 0:
            print("Fizz")
        elif x % 5 == 0:
            print("Buzz")
        else:
            print(x)

    def fizzbuzz2(x):
        res =""
        if x % 15 == 0:
            res ="FizzBuzz"
        elif x % 3 == 0:
            res = "Fizz"
        elif x % 5 == 0:
            res = "Buzz"
        else:
            res =x
        return res      

    for i in range(1, 42):
        fizzbuzz(i)

    print("===")
    print(list(map(fizzbuzz, range(1, 42))))
    print(list(map(fizzbuzz2, range(1, 42))))

File: 04-functions/rs/src/fizzbuzz.rs
pub fn demo() {
    println!("====");
    println!("fizzbuzz");
    println!("====");

    let fizzbuzz = |x| {
        if x % 15 == 0 {
            println!("FizzBuzz")
        } else if x % 3 == 0 {
            println!("Fizz")
        } else if x % 5 == 0 {
            println!("Buzz")
        } else {
            println!("{}", x)
        }
    };

    let fizzbuzz2 = |x: i32| {
        if x % 15 == 0 {
            return "FizzBuzz".to_string();
        } else if x % 3 == 0 {
            return "Fizz".to_string();
        } else if x % 5 == 0 {
            return "Buzz".to_string();
        } else {
            return x.to_string();
        }
    };
    for i in 1..42 {
        fizzbuzz(i)
    }
    println!("====");
    (1..42).into_iter().for_each(fizzbuzz);
    println!("====");
    println!("{:?}", (1..42).map(fizzbuzz2).collect::<Vec<String>>());
}
